### PR TITLE
Refactor LearningRateScheduler

### PR DESCRIPTION
**Pull request by xihui-wu**
This [line](https://github.com/tensorflow/swift-models/blob/master/TrainingLoop/Callbacks/LearningRateScheduler.swift#L55) of code and below were taken into that file from another earlier existing file. It implemented the fundamental learning rate scheduling algorithms and two of them were used by BERT model. 

It was a good starting point to build our TrainingLoop LearningRateScheduler Callback so I took it as it is in https://github.com/tensorflow/swift-models/pull/679. However this is not the right interface for user to use, so this followup PR comes out to implement a higher level interface.

### CODE REVIEW

File: TrainingLoop/Callbacks/LearningRateScheduler.swift [hunk 1]
```diff
@@ -1,54 +1,192 @@
...
+  public var shape: ScheduleShape
+
+  /// The learning rate with which warmup ends.
+  public var endLearningRate: Float
+
+  /// The step at which warmup ends.
+  public var endStep: Float
+
+  /// Create an instance of WarmupSchedule with the curve in `shape` shape and
+  /// warmup ends with `endLearningRate` rate at `endStep` step.
```
Comment (dabrahams) [inline TrainingLoop/Callbacks/LearningRateScheduler.swift:26]:
```suggestion
  /// Creates an instance with the curve in `shape`,
  /// ending with `endLearningRate`  at `endStep`.
```

File: TrainingLoop/Callbacks/LearningRateScheduler.swift [hunk 2]
```diff
@@ -1,54 +1,192 @@
...
+/// warmup steps during training.
+public struct WarmupSchedule {
+  /// The shape of learning rate curve.
+  public var shape: ScheduleShape
+
+  /// The learning rate with which warmup ends.
+  public var endLearningRate: Float
+
+  /// The step at which warmup ends.
+  public var endStep: Float
```
Comment (dabrahams) [inline TrainingLoop/Callbacks/LearningRateScheduler.swift:23]:
```suggestion
  public var endStep: Int
```

File: TrainingLoop/Callbacks/LearningRateScheduler.swift [hunk 3]
```diff
@@ -1,54 +1,192 @@
...
+
+  /// Create an instance of WarmupSchedule with the curve in `shape` shape and
+  /// warmup ends with `endLearningRate` rate at `endStep` step.
+  public init(shape: ScheduleShape, endLearningRate: Float, endStep: Float) {
```
Comment (dabrahams) [inline TrainingLoop/Callbacks/LearningRateScheduler.swift:27]:
```suggestion
  public init(shape: ScheduleShape, end rate: Float, _ step: Int) {
```

```suggestion
  public init(shape: ScheduleShape, finalRate: Float, finalStep: Int) {
```

```suggestion
  public init(shape: ScheduleShape, endLearningRate: Float, steps: Int) {
```

```suggestion
  public init(shape: ScheduleShape, endWith rate: Float, endAt step: Int) {
```

```suggestion
  public init(shape: ScheduleShape, endRate: Float, endStep: Int) {
```

**Comment by xihui-wu** [comment]
Have a new one here https://github.com/tensorflow/swift-models/pull/711
